bundles/org.smarthomej.binding.telenot/src/main/java/org/smarthomej/binding/telenot/internal/TelenotDiscoveryService.java

```diff
@@ -0,0 +1,137 @@
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smarthomej.binding.telenot.internal.handler.TelenotBridgeHandler;
+
+/**
+ * The {@link TelenotDiscoveryService} handles discovery of devices as they are identified by the bridge handler.
+ *
+ * @author Ronny Grun - Initial contribution
+ */
+@NonNullByDefault
+public class TelenotDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private final Logger logger = LoggerFactory.getLogger(TelenotDiscoveryService.class);
+
+    private @Nullable ScheduledFuture<?> scanningJob;
+    private @Nullable TelenotBridgeHandler bridgeHandler;
+    private @Nullable ThingUID bridgeUID;
+
+    public TelenotDiscoveryService() {
+        super(DISCOVERABLE_DEVICE_TYPE_UIDS, 0, true);
+    }
+
+    @Override
+    public void setThingHandler(ThingHandler thingHandler) {
+        if (thingHandler instanceof TelenotBridgeHandler) {
+            this.bridgeHandler = (TelenotBridgeHandler) thingHandler;
+            this.bridgeUID = thingHandler.getThing().getUID();
+        }
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return DISCOVERABLE_DEVICE_TYPE_UIDS;
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return bridgeHandler;
+    }
+
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        TelenotBridgeHandler bridgeHandler = this.bridgeHandler;
+        if (bridgeHandler == null) {
+            logger.warn("Tried to scan for results but bridge handler is not set in discovery service");
+            return;
+        }
+        stopScan();
+        bridgeHandler.getUsedSecurityArea().forEach(this::buildDiscoveryResult);
+
+        // we clear all older results, they are not valid any longer and we created new results
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        scheduler.schedule(() -> {
+            startScan();
+        }, 15, TimeUnit.SECONDS);
+        /*
+         * final ScheduledFuture<?> scanningJob = this.scanningJob;
+         * if (scanningJob == null || scanningJob.isCancelled()) {
+         * this.scanningJob = scheduler.scheduleWithFixedDelay(this::startScan, 0, 20, TimeUnit.SECONDS);
+         * }
+         */
+    }
+
+    /*
+     * @Override
+     * protected void stopBackgroundDiscovery() {
+     * final ScheduledFuture<?> scanningJob = this.scanningJob;
+     * if (scanningJob != null) {
+     * scanningJob.cancel(true);
+     * this.scanningJob = null;
+     * }
+     * }
+     */
```

Comment:
You are correct! I forgot to cancel the job and make it execute more than once.

```suggestion
    protected void startBackgroundDiscovery() {
         final ScheduledFuture<?> scanningJob = this.scanningJob;
         if (scanningJob == null) {
             this.scanningJob = scheduler.scheduleWithFixedDelay(this::startScan, 0, 15, TimeUnit.SECONDS);
         }
     }

    @Override
    protected void stopBackgroundDiscovery() {
        final ScheduledFuture<?> scanningJob = this.scanningJob;
        if (scanningJob != null) {
            scanningJob.cancel(true);
            this.scanningJob = null;
       }
   }
```